File: src/backend/signaling-server/src/main/java/com/bbebig/signalingserver/domain/SignalMessage.java
package com.bbebig.signalingserver.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SignalMessage {

    private MessageType messageType;
    private String channelId;           // 체날 ID
    private String senderId;            // 보낸 사람 ID
    private String receiverId;          // 받는 사람 ID
    private Object sdp;                 // SDP
    private Object candidate;           // ICE candidate
}

File: src/android/bbebig-discod-clone/presentation/src/main/java/com/smilegate/bbebig/presentation/ui/livechat/model/SignalMessage.kt
package com.smilegate.bbebig.presentation.ui.livechat.model

import com.smilegate.bbebig.data.model.SDP
import com.smilegate.bbebig.domain.model.SignalMessageDomainModel

data class SignalMessage(
    val messageType: String,
    val channelId: String,
    val senderId: String,
    val receiverId: String? = null,
    val sdp: SDP? = null,
    val candidate: Candidate? = null,
)

fun SignalMessageDomainModel.toUiModel() = SignalMessage(
    messageType = messageType,
    channelId = channelId,
    senderId = senderId,
    receiverId = receiverId,
    sdp = sdp,
    candidate = candidate?.toUiModel(),
)

fun SignalMessage.toDomainModel() = SignalMessageDomainModel(
    messageType = messageType,
    channelId = channelId,
    senderId = senderId,
    receiverId = receiverId,
    sdp = sdp,
    candidate = candidate?.toDomainModel(),
)